File: financialManagerUI/src/app/entities/requests/page.request.ts
type SortOption = {
  fieldName: string;
  ascending: boolean;
};

type Filter = {
  key: string;
  value: string;
};

export class PageRequest {
  public take: number;
  public skip: number;
  public filters?: Filter[] = undefined;
  public sortOptions?: SortOption[] = undefined;

  constructor(
    take: number,
    skip: number,
    filterOptions?: Filter[],
    sortOptions?: SortOption[]
  ) {
    this.take = take;
    this.skip = skip;
    this.filters = filterOptions;
    this.sortOptions = sortOptions;
  }
}

File: financialManagerAPI/Entities/Requests/PageRequest.cs
using System.ComponentModel.DataAnnotations;

namespace financial_manager.Entities.Requests
{
    public class PageRequest
    {
        [Range(1, 99)]
        public int Take { get; set; }
        public int Skip { get; set; }

        public Dictionary<string, string?> Filters { get; set; } = new Dictionary<string, string?>();
        public List<SortOption> SortOptions { get; set; } = new List<SortOption>();
    }

    public class SortOption
    {
        public string FieldName { get; set; } = string.Empty;
        public bool Ascending { get; set; } = true;
    }
}
